src/rgw/rgw_swift.cc

```diff
@@ -668,12 +693,17 @@ int authenticate_temp_url(RGWRados * const store, req_state * const s)
   /* We need to verify two paths because of compliance with Swift, Tempest
    * and old versions of RadosGW. The second item will have the prefix
    * of Swift API entry point removed. */
+
+  /* XXX can we search this ONCE? */
   const size_t pos = g_conf->rgw_swift_url_prefix.find_last_not_of('/') + 1;
-  const vector<string> allowed_paths = {
-    s->info.request_uri,
-    s->info.request_uri.substr(pos + 1)
+  boost::string_ref ref_uri = s->info.request_uri;
+  const vector<boost::string_ref> allowed_paths = {
```

Comment:
In https://github.com/ceph/ceph/pull/8657/commits/fbaf683a24752b61c4596c7305473259ed7d9afa this has been addressed through `boost::container::static_vector` of `boost::string_ref`. Though, I don't expect that switching to `string_ref` for such small strings would bring substantial improvement on those platforms which offer SSO. On the other hand GCC 4.8.4 on Trusty seems to be perfectly unaware about this trick. Anyway, eradicating the `vector` should be profitable.